src/server/native_controllers/index.test.js

```diff
@@ -0,0 +1,385 @@
+    };
+    const processExitSpy = jest
+      .spyOn(process, 'exit')
+      .mockImplementation(() => {});
+    const consoleErrorSpy = jest
+      .spyOn(console, 'error')
+      .mockImplementation(() => {});
+
+    safeChildProcess.mockReturnValue({ terminating: true });
+
+    await NativeControllers.prepare(kbnServer);
+    mockSpawnedProcess.emit('exit');
+
+    expect(consoleErrorSpy).toHaveBeenCalledTimes(0);
+    expect(processExitSpy).toHaveBeenCalledTimes(0);
+
+    consoleErrorSpy.mockReset();
+    consoleErrorSpy.mockRestore();
+    processExitSpy.mockReset();
+    processExitSpy.mockRestore();
+  });
+
+  test(`passes config to nativeController.configure`, async () => {
+    const kbnServer = {
+      settings: {
+        plugins: {
+          paths: [path.resolve(VALID_FIXTURES, 'correct')],
+        },
+        path: {
+          data: '/tmp/foo'
+        }
+      },
+    };
+
+    await NativeControllers.prepare(kbnServer);
+
+    expect(mockNativeControllerConfigure).toHaveBeenCalledWith({
+      'path.data': '/tmp/foo'
+    });
+  });
+
+  test(`rejects if configure rejects`, async () => {
+    const kbnServer = {
+      settings: {
+        plugins: {
+          paths: [path.resolve(VALID_FIXTURES, 'correct')],
+        },
+        path: {
+          data: '/tmp/foo'
+        }
+      },
+    };
+
+    mockNativeControllerConfigure.mockImplementation(() => Promise.reject(new Error('test error')));
+    await expect(NativeControllers.prepare(kbnServer)).rejects.toThrowErrorMatchingSnapshot();
+  });
+});
+
+describe('#killOrStart', () => {
+  test('starts enabled plugin', async () => {
+    const mockNativeController = {
+      start: jest.fn().mockReturnValue(Promise.resolve()),
+    };
+    const kbnServer = {
+      pluginSpecs: [
+        {
+          getId: () => 'foo',
+        },
+      ],
+      disabledPluginSpecs: [],
+      nativeControllers: {
+        foo: mockNativeController,
+      },
+    };
+
+    await NativeControllers.killOrStart(kbnServer);
+
+    expect(mockNativeController.start).toHaveBeenCalledTimes(1);
+  });
+
+  test('waits for started$ before resolving killOrStart Promise', async () => {
+    expect.hasAssertions();
+
+    let resolved = false;
+    let triggerStarted;
+
+    const mockNativeController = {
+      start: jest.fn().mockImplementation(
+        () =>
+          new Promise(resolve => {
+            triggerStarted = () => {
+              resolved = true;
+              resolve();
+            };
+          })
+      ),
+    };
+    const kbnServer = {
+      pluginSpecs: [
+        {
+          getId: () => 'foo',
+        },
+      ],
+      disabledPluginSpecs: [],
+      nativeControllers: {
+        foo: mockNativeController,
+      },
+    };
+
+    NativeControllers.killOrStart(kbnServer).then(() => {
+      expect(resolved).toBe(true);
+    });
+
+    triggerStarted();
+  });
+
+  test(`doesn't throw error when enabled plugin doesn't have a nativeController`, async () => {
```

Comment:
I think this should be broken down into more tests.

The current test is technically verifying a couple things, but the test name gives the impression that it only cares about one enabled plugin without a native controller, so a future refactor could end up leaving us in a place where the actual behaviors are not completely tested.

Ideally, we'll be testing three scenarios:
1. No plugins are enabled
2. One plugin is enabled but it doesn't have a native controller
3. Multiple plugins are enabled and none of them have native controllers